src/components/Input/Input.js

```diff
@@ -100,41 +132,33 @@ Input.propTypes = {
    * attribute to the <input> element.
    */
   disabled: PropTypes.bool,
-  /**
-   * Autocomplete attribute to be passed down to the <input> element.
-   */
-  autoComplete: PropTypes.string,
   /**
    * Trigger inline styles on the component.
    */
   inline: PropTypes.bool,
   /**
    * Trigger stretch (full width) styles on the component.
    */
-  stretch: (props, propName, componentName) => {
-    if (props.inline && props.stretch) {
-      return new Error(
-        'You cannot use both inline and stretch properties at the same time.'
-      );
-    }
-
-    return PropTypes.checkPropTypes(
-      { propName: PropTypes.bool },
-      props,
-      propName,
-      componentName
-    );
-  }
+  stretch: stretchPropType,
+  /**
+   * Adds bottom margin to the input.
+   */
+  margin: PropTypes.bool,
+  /**
+   * The HTML input element to render.
+   */
+  element: PropTypes.oneOf(['input', 'textarea'])
 };
 
 Input.defaultProps = {
-  theme: standard,
+  children: null,
+  element: 'input',
   invalid: false,
   optional: false,
   disabled: false,
-  autoComplete: 'none',
   inline: false,
-  stretch: false
+  stretch: false,
+  margin: true
```

Comment:
@felixjung @connor-baer -- A question. Thinking about our non-React apps. If we have a prop like this that defaults to `true`, doesn't this mean that we should actually _invert_ how we handle the CSS in order that the defaults also apply in a consistent way?

For example.

```js
const marginCss = ({ margin }) =>
  !margin && css`
    label: input__container--no-margin;
    margin: 0;
  `;
```

Like, thinking about the CSS someone would need to type to achieve our JS defaults today, it looks like:

```html
<div className="input__container input__container--margin">
   <input name="abc">
</div>
```

And actually that kinda sucks? Why should someone who doesn't have an Icon need to also put a div around every input?

I'm not so sure if this container work around is the best solution or not, looking at it from all angles. Perhaps it's less hacky to have a fixed offset for the icon rather than requiring anyone using the vanilla CSS to wrap every single input in a div?

In summary, my two points:

- Default props should probably be inverted when applied as CSS classes to ensure their behavior is encoded into the base case.
- The input wrapper is not so neat if you have to type a div everytime you wanna use an input in a non-React app (which, honestly, is the majority of apps in our world right now)

Reply:
I agree with the inverted default props, that was definitely an oversight. In that case, should we rename `margin` to `noMargin`? (@felixjung's suggestion).

About the input wrapper ... hm ... hm ... 
I've given this a lot of thought (probably way more than I should) and this is the cleanest solution I could come up with. It's optimized for CSS-in-JS because that's how I expect our components to be mainly used. It doesn't make sense to me to optimize for a use case that we intend to phase out anyway. Happy to be convinced otherwise though 😊

I just had a look at Bootstrap: they also wrap all inputs in a `div` and apply the `margin`,  `display` and `width` to the wrapper. However, they also use a separate "prepended/appended" element for icons and buttons. See the username field in [this example](https://getbootstrap.com/docs/4.0/components/forms/#inline-forms).

Reply:
True it doesn't make sense to optimize for it, but the difference is between simply having a fixed offset for the icon or this whole wrapper? Or maybe you can explain to me all the issues so I can get a clearer idea. Otherwise, I would compromise on "principles" and go for `top: 24px` on an icon rather than wrapping every single input in a `div`.

Keep in mind how many projects we have that aren't in React -- some of which will never be. 

That said - we still don't know what our validations are gonna look like, so perhaps we need a div wrapper for positioning anyways and my whole argument goes out the window 😂 

While I think `noMargin` is not as nice as a prop name, I guess it makes things predictable when it comes to the CSS classes. I think we can do camelCase for props and kebab-case for names though. So `<Input noMargin={true}/>` and `<div className="input__container--no-margin">`.